In [103]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [104]:
# CRM Tree Sizes
crm_tree_size_fp = r'E:/TreeSizes/Data/report1565196070499.csv'
crm_tree_size = pd.read_csv(crm_tree_size_fp, sep=',')

# Complete Tree Size Table
tree_size_fp = r'E:/TreeSizes/Data/complete_tree_size_list.csv'
sizes = pd.read_csv(tree_size_fp, sep=',')

# GIS Data
gis_fp = r'E:/TreeSizes/Data/DB_Metro_Grave_Trees.csv'
gis = pd.read_csv(gis_fp, sep=',')


# Tree Size Table
#tree_size_fp2 = r'E:/TreeSizes/Data/Tree Names_4.9.2018.csv'
#tree_size2 = pd.read_csv(tree_size_fp2, sep=',')


In [105]:
sizes

genus_species         crm_tree_id  \
0                     Abies concolor  a0o1I000001VWDnQAO   
1                  Acer buergerianum  a0o1I000001VW8HQAW   
2                     Acer campestre  a0o1I000001VW8IQAW   
3                       Acer ginnala  a0o1I000001VW8JQAW   
4                       Acer griseum  a0o1I000001VW8KQAW   
5                       Acer miyabei  a0o1I000001VW8LQAW   
6                       Acer negundo  a0o1I000001VWEmQAO   
7                      Acer palmatum  a0o1I000001VW8MQAW   
8                   Acer platanoides  a0o1I000001VW8NQAW   
9                        Acer rubrum  a0o1I000001VW8OQAW   
10                       Acer rubrum  a0o1I000001VW8PQAW   
11                       Acer rubrum  a0o1I000001VW8QQAW   
12                       Acer rubrum  a0o1I000001VW8SQAW   
13                       Acer rubrum  a0o1I000001VW8TQAW   
14                       Acer rubrum  a0o1I000001VW8UQAW   
15                       Acer rubrum  a0o1I000001VW8VQAW   
16                       Acer rubrum  a0o1I000001VW8WQAW   
17                       Acer rubrum  a0o1I000001VWCRQA4   
18                  Acer saccharinum  a0o1I000001VW8XQAW   
19                    Acer saccharum  a0o1I000001VWDyQAO   
20                    Acer saccharum  a0o1I000001VW8RQAW   
21                    Acer saccharum  a0o1I000001VW8YQAW   
22                    Acer saccharum  a0o1I000001VW8ZQAW   
23                    Acer saccharum  a0o1I000001VW8aQAG   
24                    Acer saccharum  a0o1I000001VWCSQA4   
25                    Acer truncatum  a0o1I000001VW8bQAG   
26                  Acer x freemanii  a0o1I000001VWEAQA4   
27                  Acer x freemanii  a0o1I000001VW8cQAG   
28                  Acer x freemanii  a0o1I000001VW8dQAG   
29                  Acer x freemanii  a0o1I000001VW8eQAG   
..                               ...                 ...   
454                 Tsuga canadensis  a0o1I000001VWBbQAO   
455                  Ulmus americana  a0o1I00000YHEhQQAX   
456                  Ulmus americana  a0o1I000001VWDZQA4   
457                  Ulmus americana  a0o1I000001VWEUQA4   
458                  Ulmus americana  a0o1I000001VWBcQAO   
459                  Ulmus americana  a0o1I000001VWBdQAO   
460                  Ulmus americana  a0o1I000001VWBeQAO   
461                  Ulmus americana  a0o1I000001VWBfQAO   
462  Ulmus carpinifolia x parvifolia  a0o1I000001VWBgQAO   
463    Ulmus davidiana var. japonica  a0o1I000001VWDfQAO   
464                     Ulmus glabra  a0o1I000001VWEBQA4   
465          Ulmus japonica x pumila  a0o1I000001VWBmQAO   
466      Ulmus japonica x wilsoniana  a0o1I000001VWBhQAO   
467      Ulmus japonica x wilsoniana  a0o1I000001VWBiQAO   
468                 Ulmus parvifolia  a0o1I000001VWBjQAO   
469                 Ulmus propinquia  a0o1I000001VWBkQAO   
470                 Ulmus wilsoniana  a0o1I000001VWBlQAO   
471                      Ulmus x sp.  a0o1I000001VWDYQA4   
472                      Ulmus x sp.  a0o1I000001VWDiQAO   
473                      Ulmus x sp.  a0o1I000001VWBnQAO   
474          Vaccinium angustifolium  a0o1I000001VWBoQAO   
475                Viburnum dentatum  a0o1I000001VWBpQAO   
476                 Viburnum lentago  a0o1I000001VWBqQAO   
477                   Viburnum nudum  a0o1I000001VWBrQAO   
478                  Viburnum opulus  a0o1I000001VWBsQAO   
479             Viburnum prunifolium  a0o1I000001VWBtQAO   
480                Viburnum trilobum  a0o1I000001VWEPQA4   
481                  Zelkova serrata  a0o1I000001VWE6QAO   
482                  Zelkova serrata  a0o1I000001VWBuQAO   
483                  Zelkova serrata  a0o1I00000OOh2kQAD   

                                scientific_name  \
0                                Abies concolor   
1                             Acer buergerianum   
2              Acer campestre - Queen Elizabeth   
3                                  Acer ginnala   
4                                  Acer griseum

In [106]:
crm_tree_size

Tree Record: Tree Record Number  Quantity Tree Species: Mature Tree Size  \
0                CRM Tree #190005472         1                            NaN   
1                CRM Tree #190005513         2                            NaN   
2                CRM Tree #190003978         2                            NaN   
3                CRM Tree #190009881         1                            NaN   
4                CRM Tree #190004221         2                            NaN   
5                CRM Tree #190003684         4                            NaN   
6                CRM Tree #190004244         1                            NaN   
7                CRM Tree #190004017         2                            NaN   
8                CRM Tree #190003875         1                            NaN   
9                CRM Tree #190002881         3                            NaN   
10               CRM Tree #190003090         2                            NaN   
11               CRM Tree #190002721         1                            NaN   
12               CRM Tree #190002280         1                            NaN   
13               CRM Tree #190002281         1                            NaN   
14               CRM Tree #190002282         1                            NaN   
15               CRM Tree #190002283         1                            NaN   
16               CRM Tree #190001751         1                            NaN   
17               CRM Tree #190001752         1                            NaN   
18               CRM Tree #190001753         1                            NaN   
19               CRM Tree #190001549         1                            NaN   
20               CRM Tree #190001550         1                            NaN   
21               CRM Tree #190001261         1                            NaN   
22               CRM Tree #190001212         1                            NaN   
23               CRM Tree #190009876         3                            NaN   
24               CRM Tree #190009202         2                            NaN   
25               CRM Tree #190004274         1                            NaN   
26               CRM Tree #190004018         2                            NaN   
27               CRM Tree #190003894         2                            NaN   
28               CRM Tree #190004799         1                            NaN   
29               CRM Tree #190003513         1                            NaN   
...                              ...       ...                            ...   
7564             CRM Tree #190002221         1                     Large Tree   
7565             CRM Tree #190002222         1                     Large Tree   
7566             CRM Tree #190002223         1                     Large Tree   
7567             CRM Tree #190002199         1                     Large Tree   
7568             CRM Tree #190002200         1                     Large Tree   
7569             CRM Tree #190002224         1                     Large Tree   
7570             CRM Tree #190002225         1                     Large Tree   
7571             CRM Tree #190002226         1                     Large Tree   
7572             CRM Tree #190002227         1                     Large Tree   
7573             CRM Tree #190002228         1                     Large Tree   
7574             CRM Tree #190001748         1                     Large Tree   
7575             CRM Tree #190001749         1                     Large Tree   
7576             CRM Tree #190001281         1                     Large Tree   
7577             CRM Tree #190008209         5                     Large Tree   
7578             CRM Tree #190008535         5                     Large Tree   
7579             CRM Tree #190009020         9                     Large Tree   
7580             CRM Tree #190008190         2                     Large Tree   
7581             CRM Tree #190008574

In [107]:
sizes.columns

Index(['genus_species', 'crm_tree_id', 'scientific_name', 'common_name',
       'size', 'priority'],
      dtype='object')

In [108]:
sciname_size = sizes.drop(['genus_species', 'crm_tree_id',  'common_name','priority'], axis=1)
sciname_size.rename(columns ={'scientific_name':'SciName'},inplace=True)
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(sciname_size)

                                               SciName         size
0                                       Abies concolor   Large Tree
1                                    Acer buergerianum   Small Tree
2                     Acer campestre - Queen Elizabeth  Medium Tree
3                                         Acer ginnala   Small Tree
4                                         Acer griseum   Small Tree
5                          Acer miyabei - State Street   Small Tree
6                             Acer negundo - Sensation   Small Tree
7                                        Acer palmatum   Small Tree
8                                     Acer platanoides   Large Tree
9                                          Acer rubrum  Medium Tree
10                             Acer rubrum - Armstrong  Medium Tree
11                       Acer rubrum - Autumn Radiance  Medium Tree
12                               Acer rubrum - Karpick  Medium Tree
13                             Acer rubrum - Nor

In [109]:
sciname_size.drop_duplicates(subset='SciName', keep='first', inplace=True)
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(sciname_size)

                                               SciName         size
0                                       Abies concolor   Large Tree
1                                    Acer buergerianum   Small Tree
2                     Acer campestre - Queen Elizabeth  Medium Tree
3                                         Acer ginnala   Small Tree
4                                         Acer griseum   Small Tree
5                          Acer miyabei - State Street   Small Tree
6                             Acer negundo - Sensation   Small Tree
7                                        Acer palmatum   Small Tree
8                                     Acer platanoides   Large Tree
9                                          Acer rubrum  Medium Tree
10                             Acer rubrum - Armstrong  Medium Tree
11                       Acer rubrum - Autumn Radiance  Medium Tree
12                               Acer rubrum - Karpick  Medium Tree
13                             Acer rubrum - Nor

In [110]:
sciname_size.groupby('size').count()

SciName
size                
Large Tree       164
Medium Tree       97
Shrub              2
Small Tree       192
shrub             29

In [111]:
sciname_size.replace('shrub','Shrub', inplace=True)

In [112]:
gis

CT_Tag_ID Legacy_ID Loc_Status           Date_Ptd         Date_Entrd  \
0           LGCY1         1      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
1           LGCY2         2      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
2           LGCY3         3      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
3           LGCY4         4      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
4           LGCY5         5      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
5           LGCY6         6      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
6           LGCY7         7      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
7           LGCY8         8      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
8           LGCY9         9      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
9          LGCY10        10      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
10         LGCY11        11      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
11         LGCY12        12      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
12         LGCY13        13      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
13         LGCY14        14      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
14         LGCY15        15      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
15         LGCY16        16      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
16         LGCY17        17      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
17         LGCY18        18      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
18         LGCY19        19      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
19         LGCY20        20      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
20         LGCY21        21      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
21         LGCY22        22      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
22         LGCY23        23      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
23         LGCY24        24      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
24         LGCY25        25      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
25         LGCY26        26      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
26         LGCY27        27      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
27         LGCY28        28      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
28         LGCY29        29      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
29         LGCY30        30      Metro  1/28/2010 0:00:00  1/11/2012 0:00:00   
...           ...       ...        ...                ...                ...   
34200  CT19000917       NaN      Metro  6/21/2019 0:00:00   7/1/2019 0:00:00   
34201  CT19000846       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34202  CT19000918       NaN      Metro  6/21/2019 0:00:00  6/21/2019 0:00:00   
34203  CT19000919       NaN      Metro  6/21/2019 0:00:00  6/21/2019 0:00:00   
34204  CT19000847       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34205  CT19000920       NaN      Metro  6/21/2019 0:00:00  6/21/2019 0:00:00   
34206  CT19000848       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34207  CT19000923       NaN      Metro  6/21/2019 0:00:00   7/1/2019 0:00:00   
34208  CT19000924       NaN      Metro  6/21/2019 0:00:00   7/1/2019 0:00:00   
34209  CT19000925       NaN      Metro  6/21/2019 0:00:00   7/1/2019 0:00:00   
34210  CT19000926       NaN      Metro  6/21/2019 0:00:00  6/21/2019 0:00:00   
34211  CT19000927       NaN      Metro  6/21/2019 0:00:00  6/21/2019 0:00:00   
34212  CT19000849       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34213  CT19000850       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34214  CT19000851       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34215  CT19000852       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34216  CT19000853       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34217  CT19000854       NaN      Metro  6/15/2019 0:00:00  6/18/2019 0:00:00   
34218  C

In [113]:
uniquscinames_gis = gis.groupby(['SciName']).nunique()
len(uniquscinames_gis)

510

In [114]:
uniquscinames_crm = sciname_size.groupby(['SciName']).nunique()
len(uniquscinames_crm)

484

In [115]:
gis.columns

Index(['CT_Tag_ID', 'Legacy_ID', 'Loc_Status', 'Date_Ptd', 'Date_Entrd',
       'Plant_User', 'Genus', 'Species', 'Cultivar', 'SciName', 'CmmnName',
       'DBH', 'StockType', 'Nursery', 'InvoiceDate', 'InvoiceNum',
       'TotalPrice', 'UnitPrice', 'Freight', 'Royalties', 'Misc_Chrg',
       'Class_Code', 'Donor_Memo', 'Expns_Code', 'EventName', 'EventDate',
       'SpaceType', 'Jurisdictn', 'CntyState', 'Program', 'Fund_Type', 'DDOE',
       'InHonorOf', 'PTG_Notes', 'Invty_Date', 'Mntc_Condn', 'Invty_User',
       'Last_Prune_Date', 'Last_Water_Date', 'Prune_Type', 'Removed',
       'GlobalID', 'SummerQuery', 'created_user', 'created_date',
       'last_edited_user', 'last_edited_date', 'TreeRecID', 'OBJECTID', 'Year',
       'Season', 'Location'],
      dtype='object')

In [116]:
trees = gis.drop([ 'Legacy_ID', 'Date_Entrd','Plant_User','DBH', 'StockType', 'Nursery', 'InvoiceDate', 'InvoiceNum',
       'TotalPrice', 'UnitPrice', 'Freight', 'Royalties', 'Misc_Chrg','Donor_Memo', 'Expns_Code','InHonorOf', 'PTG_Notes', 'Invty_Date', 'Mntc_Condn', 'Invty_User',
       'Last_Prune_Date', 'Last_Water_Date', 'Prune_Type', 'Removed','EventDate','Location','GlobalID', 'SummerQuery', 'created_user', 'created_date',
       'last_edited_user', 'last_edited_date', 'TreeRecID', 'OBJECTID'], axis=1)
trees

CT_Tag_ID Loc_Status           Date_Ptd         Genus  \
0           LGCY1      Metro  1/28/2010 0:00:00         Ulmus   
1           LGCY2      Metro  1/28/2010 0:00:00         Ulmus   
2           LGCY3      Metro  1/28/2010 0:00:00         Ulmus   
3           LGCY4      Metro  1/28/2010 0:00:00         Ulmus   
4           LGCY5      Metro  1/28/2010 0:00:00         Ulmus   
5           LGCY6      Metro  1/28/2010 0:00:00         Ulmus   
6           LGCY7      Metro  1/28/2010 0:00:00         Ulmus   
7           LGCY8      Metro  1/28/2010 0:00:00         Ulmus   
8           LGCY9      Metro  1/28/2010 0:00:00         Ulmus   
9          LGCY10      Metro  1/28/2010 0:00:00         Ulmus   
10         LGCY11      Metro  1/28/2010 0:00:00         Ulmus   
11         LGCY12      Metro  1/28/2010 0:00:00         Ulmus   
12         LGCY13      Metro  1/28/2010 0:00:00         Ulmus   
13         LGCY14      Metro  1/28/2010 0:00:00         Ulmus   
14         LGCY15      Metro  1/28/2010 0:00:00         Ulmus   
15         LGCY16      Metro  1/28/2010 0:00:00         Ulmus   
16         LGCY17      Metro  1/28/2010 0:00:00         Ulmus   
17         LGCY18      Metro  1/28/2010 0:00:00         Ulmus   
18         LGCY19      Metro  1/28/2010 0:00:00         Ulmus   
19         LGCY20      Metro  1/28/2010 0:00:00         Ulmus   
20         LGCY21      Metro  1/28/2010 0:00:00         Ulmus   
21         LGCY22      Metro  1/28/2010 0:00:00         Ulmus   
22         LGCY23      Metro  1/28/2010 0:00:00         Ulmus   
23         LGCY24      Metro  1/28/2010 0:00:00         Ulmus   
24         LGCY25      Metro  1/28/2010 0:00:00         Ulmus   
25         LGCY26      Metro  1/28/2010 0:00:00         Ulmus   
26         LGCY27      Metro  1/28/2010 0:00:00         Ulmus   
27         LGCY28      Metro  1/28/2010 0:00:00         Ulmus   
28         LGCY29      Metro  1/28/2010 0:00:00         Ulmus   
29         LGCY30      Metro  1/28/2010 0:00:00         Ulmus   
...           ...        ...                ...           ...   
34200  CT19000917      Metro  6/21/2019 0:00:00        Ginkgo   
34201  CT19000846      Metro  6/15/2019 0:00:00       Quercus   
34202  CT19000918      Metro  6/21/2019 0:00:00     Gleditsia   
34203  CT19000919      Metro  6/21/2019 0:00:00     Gleditsia   
34204  CT19000847      Metro  6/15/2019 0:00:00       Quercus   
34205  CT19000920      Metro  6/21/2019 0:00:00  Liriodendron   
34206  CT19000848      Metro  6/15/2019 0:00:00       Quercus   
34207  CT19000923      Metro  6/21/2019 0:00:00       Quercus   
34208  CT19000924      Metro  6/21/2019 0:00:00      Taxodium   
34209  CT19000925      Metro  6/21/2019 0:00:00      Taxodium   
34210  CT19000926      Metro  6/21/2019 0:00:00         Tilia   
34211  CT19000927      Metro  6/21/2019 0:00:00         Malus   
34212  CT19000849      Metro  6/15/2019 0:00:00       Quercus   
34213  CT19000850      Metro  6/15/2019 0:00:00       Quercus   
34214  CT19000851      Metro  6/15/2019 0:00:00       Quercus   
34215  CT19000852      Metro  6/15/2019 0:00:00       Quercus   
34216  CT19000853      Metro  6/15/2019 0:00:00       Quercus   
34217  CT19000854      Metro  6/15/2019 0:00:00       Quercus   
34218  CT19000855      Metro  6/15/2019 0:00:00       Robinia   
34219  CT19000856      Metro  6/15/2019 0:00:00       Robinia   
34220  CT19000857      Metro  6/15/2019 0:00:00       Robinia   
34221  CT19000858      Metro  6/15/2019 0:00:00       Robinia   
34222  CT19000859      Metro  6/15/2019 0:00:00       Robinia   
34223  CT19000860      Metro  6/15/2019 0:00:00       Robinia   
34224  CT19000861      Metro  6/15/2019 0:00:00       Robinia   
34225  CT19000862      Metro  6/15/2019 0:00:00       Robinia   
34226  CT19000863      Metro  6/15/2019 0:00:00       Robinia   
34227  CT19000864      Metro  6/15/2019 0:00:00      Taxodium   
34228  CT19000865      Metro  6/15/2019 0:00:00      Taxodium   
34229  CT19000866      Metro  6/15/2019 0:0

In [117]:
dbtree_sizes = pd.merge(trees, sciname_size, how='left', on='SciName')
dbtree_sizes

CT_Tag_ID Loc_Status           Date_Ptd         Genus  \
0           LGCY1      Metro  1/28/2010 0:00:00         Ulmus   
1           LGCY2      Metro  1/28/2010 0:00:00         Ulmus   
2           LGCY3      Metro  1/28/2010 0:00:00         Ulmus   
3           LGCY4      Metro  1/28/2010 0:00:00         Ulmus   
4           LGCY5      Metro  1/28/2010 0:00:00         Ulmus   
5           LGCY6      Metro  1/28/2010 0:00:00         Ulmus   
6           LGCY7      Metro  1/28/2010 0:00:00         Ulmus   
7           LGCY8      Metro  1/28/2010 0:00:00         Ulmus   
8           LGCY9      Metro  1/28/2010 0:00:00         Ulmus   
9          LGCY10      Metro  1/28/2010 0:00:00         Ulmus   
10         LGCY11      Metro  1/28/2010 0:00:00         Ulmus   
11         LGCY12      Metro  1/28/2010 0:00:00         Ulmus   
12         LGCY13      Metro  1/28/2010 0:00:00         Ulmus   
13         LGCY14      Metro  1/28/2010 0:00:00         Ulmus   
14         LGCY15      Metro  1/28/2010 0:00:00         Ulmus   
15         LGCY16      Metro  1/28/2010 0:00:00         Ulmus   
16         LGCY17      Metro  1/28/2010 0:00:00         Ulmus   
17         LGCY18      Metro  1/28/2010 0:00:00         Ulmus   
18         LGCY19      Metro  1/28/2010 0:00:00         Ulmus   
19         LGCY20      Metro  1/28/2010 0:00:00         Ulmus   
20         LGCY21      Metro  1/28/2010 0:00:00         Ulmus   
21         LGCY22      Metro  1/28/2010 0:00:00         Ulmus   
22         LGCY23      Metro  1/28/2010 0:00:00         Ulmus   
23         LGCY24      Metro  1/28/2010 0:00:00         Ulmus   
24         LGCY25      Metro  1/28/2010 0:00:00         Ulmus   
25         LGCY26      Metro  1/28/2010 0:00:00         Ulmus   
26         LGCY27      Metro  1/28/2010 0:00:00         Ulmus   
27         LGCY28      Metro  1/28/2010 0:00:00         Ulmus   
28         LGCY29      Metro  1/28/2010 0:00:00         Ulmus   
29         LGCY30      Metro  1/28/2010 0:00:00         Ulmus   
...           ...        ...                ...           ...   
34200  CT19000917      Metro  6/21/2019 0:00:00        Ginkgo   
34201  CT19000846      Metro  6/15/2019 0:00:00       Quercus   
34202  CT19000918      Metro  6/21/2019 0:00:00     Gleditsia   
34203  CT19000919      Metro  6/21/2019 0:00:00     Gleditsia   
34204  CT19000847      Metro  6/15/2019 0:00:00       Quercus   
34205  CT19000920      Metro  6/21/2019 0:00:00  Liriodendron   
34206  CT19000848      Metro  6/15/2019 0:00:00       Quercus   
34207  CT19000923      Metro  6/21/2019 0:00:00       Quercus   
34208  CT19000924      Metro  6/21/2019 0:00:00      Taxodium   
34209  CT19000925      Metro  6/21/2019 0:00:00      Taxodium   
34210  CT19000926      Metro  6/21/2019 0:00:00         Tilia   
34211  CT19000927      Metro  6/21/2019 0:00:00         Malus   
34212  CT19000849      Metro  6/15/2019 0:00:00       Quercus   
34213  CT19000850      Metro  6/15/2019 0:00:00       Quercus   
34214  CT19000851      Metro  6/15/2019 0:00:00       Quercus   
34215  CT19000852      Metro  6/15/2019 0:00:00       Quercus   
34216  CT19000853      Metro  6/15/2019 0:00:00       Quercus   
34217  CT19000854      Metro  6/15/2019 0:00:00       Quercus   
34218  CT19000855      Metro  6/15/2019 0:00:00       Robinia   
34219  CT19000856      Metro  6/15/2019 0:00:00       Robinia   
34220  CT19000857      Metro  6/15/2019 0:00:00       Robinia   
34221  CT19000858      Metro  6/15/2019 0:00:00       Robinia   
34222  CT19000859      Metro  6/15/2019 0:00:00       Robinia   
34223  CT19000860      Metro  6/15/2019 0:00:00       Robinia   
34224  CT19000861      Metro  6/15/2019 0:00:00       Robinia   
34225  CT19000862      Metro  6/15/2019 0:00:00       Robinia   
34226  CT19000863      Metro  6/15/2019 0:00:00       Robinia   
34227  CT19000864      Metro  6/15/2019 0:00:00      Taxodium   
34228  CT19000865      Metro  6/15/2019 0:00:00      Taxodium   
34229  CT19000866      Metro  6/15/2019 0:0

In [118]:
dbtree_sizes.isnull().sum()

CT_Tag_ID         1
Loc_Status        0
Date_Ptd       1690
Genus            45
Species          18
Cultivar      11035
SciName           2
CmmnName         31
Class_Code     1610
EventName      1714
SpaceType      2928
Jurisdictn     2568
CntyState       310
Program        2810
Fund_Type     19912
DDOE          14577
Year           1753
Season         1752
size            505
dtype: int64

In [119]:
grouping = dbtree_sizes.groupby(['size'])
size_sum = grouping.describe()
size_sum

Year                                                         \
               count         mean       std     min     25%     50%     75%   
size                                                                          
Large Tree   12715.0  2012.739127  4.442665  2003.0  2010.0  2014.0  2016.0   
Medium Tree   9139.0  2013.490644  3.775112  2003.0  2011.0  2014.0  2017.0   
Shrub          619.0  2015.001616  2.236429  2005.0  2014.0  2015.0  2017.0   
Small Tree    9508.0  2013.702040  4.037154  2003.0  2011.0  2014.0  2017.0   

                     
                max  
size                 
Large Tree   2019.0  
Medium Tree  2019.0  
Shrub        2019.0  
Small Tree   2081.0

In [120]:
dbtree_sizes.groupby('size').count()

CT_Tag_ID  Loc_Status  Date_Ptd  Genus  Species  Cultivar  \
size                                                                     
Large Tree       13214       13214     12737  13214    13214      9094   
Medium Tree       9399        9399      9158   9399     9399      6227   
Shrub             1140        1140       619   1140     1140       832   
Small Tree        9972        9972      9528   9958     9972      6630   

             SciName  CmmnName  Class_Code  EventName  SpaceType  Jurisdictn  \
size                                                                           
Large Tree     13214     13192       12758      12743      12335       12447   
Medium Tree     9399      9398        9189       9131       9004        9121   
Shrub           1140      1140         632        649        398         398   
Small Tree      9972      9967        9542       9495       9064        9195   

             CntyState  Program  Fund_Type  DDOE   Year  Season  
size                                                             
Large Tree       13114    12334       4761  7848  12715   12715  
Medium Tree       9308     9062       4727  5775   9139    9140  
Shrub             1128      397        127    58    619     619  
Small Tree        9866     9125       4487  5666   9508    9508

In [121]:
#v = (9360 + 8647 + 6952 + 5294)
v = len(dbtree_sizes)
l = len(trees)
missing_values = l - v
print(v)
print(l)
print(missing_values)

34230
34230
0


In [122]:
dbtree_sizes['size'].fillna('Unknown', inplace=True)

In [123]:
dbtree_sizes.loc[dbtree_sizes['size'] == 'NA']

Empty DataFrame
Columns: [CT_Tag_ID, Loc_Status, Date_Ptd, Genus, Species, Cultivar, SciName, CmmnName, Class_Code, EventName, SpaceType, Jurisdictn, CntyState, Program, Fund_Type, DDOE, Year, Season, size]
Index: []

In [124]:
grouping = dbtree_sizes.groupby(['size'])
size_sum = grouping.describe()
size_sum

Year                                                         \
               count         mean       std     min     25%     50%     75%   
size                                                                          
Large Tree   12715.0  2012.739127  4.442665  2003.0  2010.0  2014.0  2016.0   
Medium Tree   9139.0  2013.490644  3.775112  2003.0  2011.0  2014.0  2017.0   
Shrub          619.0  2015.001616  2.236429  2005.0  2014.0  2015.0  2017.0   
Small Tree    9508.0  2013.702040  4.037154  2003.0  2011.0  2014.0  2017.0   
Unknown        496.0  2013.471774  4.798065  2004.0  2011.0  2015.0  2018.0   

                     
                max  
size                 
Large Tree   2019.0  
Medium Tree  2019.0  
Shrub        2019.0  
Small Tree   2081.0  
Unknown      2019.0

In [140]:
indexNames = dbtree_sizes[dbtree_sizes['Class_Code'] == 'Test' ].index 
dbtree_sizes.drop(indexNames , inplace=True)
dbtree_sizes

CT_Tag_ID Loc_Status           Date_Ptd         Genus  \
0           LGCY1      Metro  1/28/2010 0:00:00         Ulmus   
1           LGCY2      Metro  1/28/2010 0:00:00         Ulmus   
2           LGCY3      Metro  1/28/2010 0:00:00         Ulmus   
3           LGCY4      Metro  1/28/2010 0:00:00         Ulmus   
4           LGCY5      Metro  1/28/2010 0:00:00         Ulmus   
5           LGCY6      Metro  1/28/2010 0:00:00         Ulmus   
6           LGCY7      Metro  1/28/2010 0:00:00         Ulmus   
7           LGCY8      Metro  1/28/2010 0:00:00         Ulmus   
8           LGCY9      Metro  1/28/2010 0:00:00         Ulmus   
9          LGCY10      Metro  1/28/2010 0:00:00         Ulmus   
10         LGCY11      Metro  1/28/2010 0:00:00         Ulmus   
11         LGCY12      Metro  1/28/2010 0:00:00         Ulmus   
12         LGCY13      Metro  1/28/2010 0:00:00         Ulmus   
13         LGCY14      Metro  1/28/2010 0:00:00         Ulmus   
14         LGCY15      Metro  1/28/2010 0:00:00         Ulmus   
15         LGCY16      Metro  1/28/2010 0:00:00         Ulmus   
16         LGCY17      Metro  1/28/2010 0:00:00         Ulmus   
17         LGCY18      Metro  1/28/2010 0:00:00         Ulmus   
18         LGCY19      Metro  1/28/2010 0:00:00         Ulmus   
19         LGCY20      Metro  1/28/2010 0:00:00         Ulmus   
20         LGCY21      Metro  1/28/2010 0:00:00         Ulmus   
21         LGCY22      Metro  1/28/2010 0:00:00         Ulmus   
22         LGCY23      Metro  1/28/2010 0:00:00         Ulmus   
23         LGCY24      Metro  1/28/2010 0:00:00         Ulmus   
24         LGCY25      Metro  1/28/2010 0:00:00         Ulmus   
25         LGCY26      Metro  1/28/2010 0:00:00         Ulmus   
26         LGCY27      Metro  1/28/2010 0:00:00         Ulmus   
27         LGCY28      Metro  1/28/2010 0:00:00         Ulmus   
28         LGCY29      Metro  1/28/2010 0:00:00         Ulmus   
29         LGCY30      Metro  1/28/2010 0:00:00         Ulmus   
...           ...        ...                ...           ...   
34200  CT19000917      Metro  6/21/2019 0:00:00        Ginkgo   
34201  CT19000846      Metro  6/15/2019 0:00:00       Quercus   
34202  CT19000918      Metro  6/21/2019 0:00:00     Gleditsia   
34203  CT19000919      Metro  6/21/2019 0:00:00     Gleditsia   
34204  CT19000847      Metro  6/15/2019 0:00:00       Quercus   
34205  CT19000920      Metro  6/21/2019 0:00:00  Liriodendron   
34206  CT19000848      Metro  6/15/2019 0:00:00       Quercus   
34207  CT19000923      Metro  6/21/2019 0:00:00       Quercus   
34208  CT19000924      Metro  6/21/2019 0:00:00      Taxodium   
34209  CT19000925      Metro  6/21/2019 0:00:00      Taxodium   
34210  CT19000926      Metro  6/21/2019 0:00:00         Tilia   
34211  CT19000927      Metro  6/21/2019 0:00:00         Malus   
34212  CT19000849      Metro  6/15/2019 0:00:00       Quercus   
34213  CT19000850      Metro  6/15/2019 0:00:00       Quercus   
34214  CT19000851      Metro  6/15/2019 0:00:00       Quercus   
34215  CT19000852      Metro  6/15/2019 0:00:00       Quercus   
34216  CT19000853      Metro  6/15/2019 0:00:00       Quercus   
34217  CT19000854      Metro  6/15/2019 0:00:00       Quercus   
34218  CT19000855      Metro  6/15/2019 0:00:00       Robinia   
34219  CT19000856      Metro  6/15/2019 0:00:00       Robinia   
34220  CT19000857      Metro  6/15/2019 0:00:00       Robinia   
34221  CT19000858      Metro  6/15/2019 0:00:00       Robinia   
34222  CT19000859      Metro  6/15/2019 0:00:00       Robinia   
34223  CT19000860      Metro  6/15/2019 0:00:00       Robinia   
34224  CT19000861      Metro  6/15/2019 0:00:00       Robinia   
34225  CT19000862      Metro  6/15/2019 0:00:00       Robinia   
34226  CT19000863      Metro  6/15/2019 0:00:00       Robinia   
34227  CT19000864      Metro  6/15/2019 0:00:00      Taxodium   
34228  CT19000865      Metro  6/15/2019 0:00:00      Taxodium   
34229  CT19000866      Metro  6/15/2019 0:0

In [151]:
dbtree_sizes['Class_Code'].fillna('0', inplace=True)

In [152]:
grouping2a = dbtree_sizes.groupby(['Class_Code'])
size_sum2a = grouping2a.describe()
size_sum2a

Year                                                          \
             count         mean       std     min     25%     50%      75%   
Class_Code                                                                   
21.0         566.0  2014.049470  1.027597  2012.0  2013.0  2014.0  2015.00   
55.0         134.0  2005.261194  0.440934  2005.0  2005.0  2005.0  2006.00   
119.0       5591.0  2010.130209  2.249278  2004.0  2008.0  2011.0  2012.00   
120.0       3696.0  2006.821158  2.999269  2003.0  2004.0  2006.0  2009.00   
121.0        175.0  2005.080000  0.272072  2005.0  2005.0  2005.0  2005.00   
124.0         14.0  2007.000000  0.000000  2007.0  2007.0  2007.0  2007.00   
125.0        821.0  2008.064555  0.953542  2007.0  2007.0  2008.0  2009.00   
127.0         72.0  2010.597222  0.493899  2010.0  2010.0  2011.0  2011.00   
130.0       1933.0  2011.012416  0.838916  2010.0  2010.0  2011.0  2012.00   
132.0        662.0  2013.058912  0.758913  2012.0  2013.0  2013.0  2013.00   
134.0         38.0  2012.000000  0.000000  2012.0  2012.0  2012.0  2012.00   
135.0        551.0  2014.353902  0.478614  2014.0  2014.0  2014.0  2015.00   
136.0          8.0  2016.000000  0.000000  2016.0  2016.0  2016.0  2016.00   
138.0         96.0  2016.458333  0.500876  2016.0  2016.0  2016.0  2017.00   
139.0          6.0  2017.333333  0.516398  2017.0  2017.0  2017.0  2017.75   
140.0        851.0  2018.995300  0.096900  2017.0  2019.0  2019.0  2019.00   
155.0       1371.0  2015.622903  2.782904  2013.0  2013.0  2014.0  2019.00   
801.0        167.0  2012.191617  3.493426  2006.0  2010.0  2014.0  2014.00   
802.0        348.0  2012.313218  2.014972  2008.0  2011.0  2012.0  2013.00   
803.0        347.0  2012.219020  2.249826  2009.0  2011.0  2012.0  2014.00   
804.0         21.0  2015.000000  0.000000  2015.0  2015.0  2015.0  2015.00   
805.0         90.0  2014.322222  1.734176  2013.0  2013.0  2014.0  2015.00   
0              8.0  2013.750000  1.035098  2013.0  2013.0  2013.0  2015.00   
119         1002.0  2015.267465  0.959895  2012.0  2015.0  2015.0  2016.00   
120          252.0  2015.484127  1.325413  2013.0  2014.0  2016.0  2016.00   
125            2.0  2007.000000  0.000000  2007.0  2007.0  2007.0  2007.00   
132          978.0  2014.861963  0.614033  2013.0  2015.0  2015.0  2015.00   
135          760.0  2014.688158  0.463551  2014.0  2014.0  2015.0  2015.00   
136          147.0  2016.000000  0.000000  2016.0  2016.0  2016.0  2016.00   
138         2106.0  2016.507123  0.500068  2016.0  2016.0  2017.0  2017.00   
139          823.0  2017.589307  0.492259  2017.0  2017.0  2018.0  2018.00   
140         2910.0  2018.099313  1.277068  2017.0  2018.0  2018.0  2018.00   
155         2894.0  2016.114375  1.578584  2013.0  2015.0  2016.0  2017.00   
21           990.0  2014.164646  0.537908  2013.0  2014.0  2014.0  2014.00   
801          988.0  2015.826923  1.398872  2013.0  2015.0  2016.0  2017.00   
802          151.0  2015.841060  1.755345  2014.0  2014.0  2016.0  2017.50   
803          227.0  2017.066079  1.320379  2015.0  2016.0  2017.0  2018.00   
804          263.0  2016.205323  2.459529  2013.0  2013.0  2018.0  2018.50   
805          397.0  2016.125945  1.537168  2013.0  2015.0  2016.0  2018.00   
95            21.0  2013.000000  0.000000  2013.0  2013.0  2013.0  2013.00   

                    
               max  
Class_Code          
21.0        2016.0  
55.0        2006.0  
119.0       2016.0  
120.0       2017.0  
121.0       2006.0  
124.0       2007.0  
125.0       2011.0  
127.0       2011.0  
130.0       2012.0  
132.0       2016.0  
134.0       2012.0  
135.0       2015.0  
136.0       2016.0  
138.0       2017.0  
139.0       2018.0  
140.0       2019.0  
155.0       2019.0  
801.0       2019.0  
802.0       2019.0  
803.0       2019.0  
804.0       2015.0  
805.0       2019.0  
0           2015.0  
119         2017.0  
120         2019.0  
125         2007.0  
132         2016.0  
135         2015.0  
136     

In [153]:
dbtree_sizes.dtypes

CT_Tag_ID      object
Loc_Status     object
Date_Ptd       object
Genus          object
Species        object
Cultivar       object
SciName        object
CmmnName       object
Class_Code     object
EventName      object
SpaceType      object
Jurisdictn     object
CntyState      object
Program        object
Fund_Type      object
DDOE           object
Year          float64
Season         object
size           object
dtype: object

In [136]:
#dbtree_sizes_t(['int_cc'], dtype='integer')

TypeError: 'DataFrame' object is not callable

In [158]:
#df["a"] = pd.to_numeric(df["a"])
#dbtree_sizes_num['int_cc'] = 

#int_cc = pd.to_numeric(dbtree_sizes_num['Class_Code'])
#dbtree_sizes['Class_Code'].apply(pd.to_numeric)
#cc = dbtree_sizes['Class_Code']
#int_cc = pd.to_numeric(cc, errors='coerce', downcast='integer')

dbtree_sizes['Class_Code'].astype(int)

grouping2a = dbtree_sizes.groupby(['Class_Code'])
size_sum2a = grouping2a.describe()
size_sum2a

#dbtree_sizes

Year                                                          \
             count         mean       std     min     25%     50%      75%   
Class_Code                                                                   
21.0         566.0  2014.049470  1.027597  2012.0  2013.0  2014.0  2015.00   
55.0         134.0  2005.261194  0.440934  2005.0  2005.0  2005.0  2006.00   
119.0       5591.0  2010.130209  2.249278  2004.0  2008.0  2011.0  2012.00   
120.0       3696.0  2006.821158  2.999269  2003.0  2004.0  2006.0  2009.00   
121.0        175.0  2005.080000  0.272072  2005.0  2005.0  2005.0  2005.00   
124.0         14.0  2007.000000  0.000000  2007.0  2007.0  2007.0  2007.00   
125.0        821.0  2008.064555  0.953542  2007.0  2007.0  2008.0  2009.00   
127.0         72.0  2010.597222  0.493899  2010.0  2010.0  2011.0  2011.00   
130.0       1933.0  2011.012416  0.838916  2010.0  2010.0  2011.0  2012.00   
132.0        662.0  2013.058912  0.758913  2012.0  2013.0  2013.0  2013.00   
134.0         38.0  2012.000000  0.000000  2012.0  2012.0  2012.0  2012.00   
135.0        551.0  2014.353902  0.478614  2014.0  2014.0  2014.0  2015.00   
136.0          8.0  2016.000000  0.000000  2016.0  2016.0  2016.0  2016.00   
138.0         96.0  2016.458333  0.500876  2016.0  2016.0  2016.0  2017.00   
139.0          6.0  2017.333333  0.516398  2017.0  2017.0  2017.0  2017.75   
140.0        851.0  2018.995300  0.096900  2017.0  2019.0  2019.0  2019.00   
155.0       1371.0  2015.622903  2.782904  2013.0  2013.0  2014.0  2019.00   
801.0        167.0  2012.191617  3.493426  2006.0  2010.0  2014.0  2014.00   
802.0        348.0  2012.313218  2.014972  2008.0  2011.0  2012.0  2013.00   
803.0        347.0  2012.219020  2.249826  2009.0  2011.0  2012.0  2014.00   
804.0         21.0  2015.000000  0.000000  2015.0  2015.0  2015.0  2015.00   
805.0         90.0  2014.322222  1.734176  2013.0  2013.0  2014.0  2015.00   
0              8.0  2013.750000  1.035098  2013.0  2013.0  2013.0  2015.00   
119         1002.0  2015.267465  0.959895  2012.0  2015.0  2015.0  2016.00   
120          252.0  2015.484127  1.325413  2013.0  2014.0  2016.0  2016.00   
125            2.0  2007.000000  0.000000  2007.0  2007.0  2007.0  2007.00   
132          978.0  2014.861963  0.614033  2013.0  2015.0  2015.0  2015.00   
135          760.0  2014.688158  0.463551  2014.0  2014.0  2015.0  2015.00   
136          147.0  2016.000000  0.000000  2016.0  2016.0  2016.0  2016.00   
138         2106.0  2016.507123  0.500068  2016.0  2016.0  2017.0  2017.00   
139          823.0  2017.589307  0.492259  2017.0  2017.0  2018.0  2018.00   
140         2910.0  2018.099313  1.277068  2017.0  2018.0  2018.0  2018.00   
155         2894.0  2016.114375  1.578584  2013.0  2015.0  2016.0  2017.00   
21           990.0  2014.164646  0.537908  2013.0  2014.0  2014.0  2014.00   
801          988.0  2015.826923  1.398872  2013.0  2015.0  2016.0  2017.00   
802          151.0  2015.841060  1.755345  2014.0  2014.0  2016.0  2017.50   
803          227.0  2017.066079  1.320379  2015.0  2016.0  2017.0  2018.00   
804          263.0  2016.205323  2.459529  2013.0  2013.0  2018.0  2018.50   
805          397.0  2016.125945  1.537168  2013.0  2015.0  2016.0  2018.00   
95            21.0  2013.000000  0.000000  2013.0  2013.0  2013.0  2013.00   

                    
               max  
Class_Code          
21.0        2016.0  
55.0        2006.0  
119.0       2016.0  
120.0       2017.0  
121.0       2006.0  
124.0       2007.0  
125.0       2011.0  
127.0       2011.0  
130.0       2012.0  
132.0       2016.0  
134.0       2012.0  
135.0       2015.0  
136.0       2016.0  
138.0       2017.0  
139.0       2018.0  
140.0       2019.0  
155.0       2019.0  
801.0       2019.0  
802.0       2019.0  
803.0       2019.0  
804.0       2015.0  
805.0       2019.0  
0           2015.0  
119         2017.0  
120         2019.0  
125         2007.0  
132         2016.0  
135         2015.0  
136     

In [164]:
dbtree_sizes.groupby(['Class_Code']).count()

CT_Tag_ID  Loc_Status  Date_Ptd  Genus  Species  Cultivar  \
Class_Code                                                              
21.0              566         566       566    566      566       299   
55.0              134         134       134    134      134       133   
119.0            5591        5591      5591   5583     5582      5275   
120.0            3702        3702      3696   3701     3699      3566   
121.0             175         175       175    175      175       175   
124.0              14          14        14     14       14         0   
125.0             821         821       821    821      821       821   
127.0              72          72        72     72       72        72   
130.0            1933        1933      1933   1933     1933      1932   
132.0             662         662       662    662      662       485   
134.0              38          38        38     38       38        38   
135.0             552         552       551    552      552       267   
136.0               8           8         8      8        8         4   
138.0              96          96        96     96       96        18   
139.0               6           6         6      6        6         0   
140.0             861         861       861    861      861       324   
155.0            1409        1409      1408   1408     1408       907   
801.0             168         168       167    168      168        97   
802.0             348         348       348    348      348       147   
803.0             347         347       347    347      347       275   
804.0              21          21        21     21       21         6   
805.0              90          90        90     90       90        42   
0                1609        1610         8   1608     1606      1280   
119              1002        1002      1002   1002     1002       350   
120               324         324       254    324      324       135   
125                 2           2         2      2        2         2   
132               978         978       978    978      978       209   
135               761         761       760    761      761       353   
136               147         147       147    147      147        45   
138              2106        2106      2106   2106     2106      1088   
139               823         823       823    823      823       430   
140              2921        2921      2921   2888     2920      1234   
155              2903        2903      2897   2903     2903      1549   
21                990         990       990    990      990       747   
801               989         989       988    989      989       416   
802               151         151       151    151      151       103   
803               227         227       227    227      227       110   
804               263         263       263    263      263        65   
805               397         397       397    397      397       195   
95                 21          21        21     21       21         0   

            SciName  CmmnName  EventName  SpaceType  Jurisdictn  CntyState  \
Class_Code                                                                   
21.0            566       566        566        406         407        566   
55.0            134       134        134        134         134        134   
119.0          5591      5591       5591       5587        5591       5591   
120.0          3702      3702       3694       3696        3696       3702   
121.0           175       175        175        175         175        175   
124.0            14        14         14         14          14         14   
125.0           821       821        821        821         821        821   
127.0            72        72         72         72          72         72   
130.0          1933      1933       1933       1933        1933       1933   
132.0           662       662        662        662         662        662   
134.0    

To be combined and called other:
21, 55, 95, 121, 127, 134, 801, 803, 804, 805

Keep – 119, 120 – call Tree Planting/CTP

Combine and call Riversmart – 125, 130, 155, 

Combine and call Large Parcel – 132, 136, 139, 140, 

Combine and call Sustainable DC – 135, 138


In [172]:
def new_program (row):
#    if row['Class_Code'] == [other]:
#        return 'Other'
    if row['Class_Code'] == 119:
        return 'Tree Planting / CTP'
    if row['Class_Code'] == 120:
        return 'Tree Planting / CTP'
    if row['Class_Code'] == 125:
        return 'Riversmart'
    if row['Class_Code'] == 130:
        return 'Riversmart'
    if row['Class_Code'] == 155:
        return 'Riversmart'
    if row['Class_Code'] == 132:
        return 'Large Parcel'
    if row['Class_Code'] == 136:
        return 'Large Parcel'
    if row['Class_Code'] == 139:
        return 'Large Parcel'
    if row['Class_Code'] == 140:
        return 'Large Parcel'
    if row['Class_Code'] == 138:
        return 'Sustainable DC'
    if row['Class_Code'] == 135:
        return 'Sustainable DC'
    if row['Class_Code'] == 119.0:
        return 'Tree Planting / CTP'
    if row['Class_Code'] == 120.0:
        return 'Tree Planting / CTP'
    if row['Class_Code'] == 125.0:
        return 'Riversmart'
    if row['Class_Code'] == 130.0:
        return 'Riversmart'
    if row['Class_Code'] == 155.0:
        return 'Riversmart'
    if row['Class_Code'] == 132.0:
        return 'Large Parcel'
    if row['Class_Code'] == 136.0:
        return 'Large Parcel'
    if row['Class_Code'] == 139.0:
        return 'Large Parcel'
    if row['Class_Code'] == 140.0:
        return 'Large Parcel'
    if row['Class_Code'] == 138.0:
        return 'Sustainable DC'
    if row['Class_Code'] == 135.0:
        return 'Sustainable DC'
    return "Other"

dbtree_sizes['new_program'] = dbtree_sizes.apply(lambda row: new_program(row), axis=1 )

dbtree_sizes

CT_Tag_ID Loc_Status           Date_Ptd         Genus  \
0           LGCY1      Metro  1/28/2010 0:00:00         Ulmus   
1           LGCY2      Metro  1/28/2010 0:00:00         Ulmus   
2           LGCY3      Metro  1/28/2010 0:00:00         Ulmus   
3           LGCY4      Metro  1/28/2010 0:00:00         Ulmus   
4           LGCY5      Metro  1/28/2010 0:00:00         Ulmus   
5           LGCY6      Metro  1/28/2010 0:00:00         Ulmus   
6           LGCY7      Metro  1/28/2010 0:00:00         Ulmus   
7           LGCY8      Metro  1/28/2010 0:00:00         Ulmus   
8           LGCY9      Metro  1/28/2010 0:00:00         Ulmus   
9          LGCY10      Metro  1/28/2010 0:00:00         Ulmus   
10         LGCY11      Metro  1/28/2010 0:00:00         Ulmus   
11         LGCY12      Metro  1/28/2010 0:00:00         Ulmus   
12         LGCY13      Metro  1/28/2010 0:00:00         Ulmus   
13         LGCY14      Metro  1/28/2010 0:00:00         Ulmus   
14         LGCY15      Metro  1/28/2010 0:00:00         Ulmus   
15         LGCY16      Metro  1/28/2010 0:00:00         Ulmus   
16         LGCY17      Metro  1/28/2010 0:00:00         Ulmus   
17         LGCY18      Metro  1/28/2010 0:00:00         Ulmus   
18         LGCY19      Metro  1/28/2010 0:00:00         Ulmus   
19         LGCY20      Metro  1/28/2010 0:00:00         Ulmus   
20         LGCY21      Metro  1/28/2010 0:00:00         Ulmus   
21         LGCY22      Metro  1/28/2010 0:00:00         Ulmus   
22         LGCY23      Metro  1/28/2010 0:00:00         Ulmus   
23         LGCY24      Metro  1/28/2010 0:00:00         Ulmus   
24         LGCY25      Metro  1/28/2010 0:00:00         Ulmus   
25         LGCY26      Metro  1/28/2010 0:00:00         Ulmus   
26         LGCY27      Metro  1/28/2010 0:00:00         Ulmus   
27         LGCY28      Metro  1/28/2010 0:00:00         Ulmus   
28         LGCY29      Metro  1/28/2010 0:00:00         Ulmus   
29         LGCY30      Metro  1/28/2010 0:00:00         Ulmus   
...           ...        ...                ...           ...   
34200  CT19000917      Metro  6/21/2019 0:00:00        Ginkgo   
34201  CT19000846      Metro  6/15/2019 0:00:00       Quercus   
34202  CT19000918      Metro  6/21/2019 0:00:00     Gleditsia   
34203  CT19000919      Metro  6/21/2019 0:00:00     Gleditsia   
34204  CT19000847      Metro  6/15/2019 0:00:00       Quercus   
34205  CT19000920      Metro  6/21/2019 0:00:00  Liriodendron   
34206  CT19000848      Metro  6/15/2019 0:00:00       Quercus   
34207  CT19000923      Metro  6/21/2019 0:00:00       Quercus   
34208  CT19000924      Metro  6/21/2019 0:00:00      Taxodium   
34209  CT19000925      Metro  6/21/2019 0:00:00      Taxodium   
34210  CT19000926      Metro  6/21/2019 0:00:00         Tilia   
34211  CT19000927      Metro  6/21/2019 0:00:00         Malus   
34212  CT19000849      Metro  6/15/2019 0:00:00       Quercus   
34213  CT19000850      Metro  6/15/2019 0:00:00       Quercus   
34214  CT19000851      Metro  6/15/2019 0:00:00       Quercus   
34215  CT19000852      Metro  6/15/2019 0:00:00       Quercus   
34216  CT19000853      Metro  6/15/2019 0:00:00       Quercus   
34217  CT19000854      Metro  6/15/2019 0:00:00       Quercus   
34218  CT19000855      Metro  6/15/2019 0:00:00       Robinia   
34219  CT19000856      Metro  6/15/2019 0:00:00       Robinia   
34220  CT19000857      Metro  6/15/2019 0:00:00       Robinia   
34221  CT19000858      Metro  6/15/2019 0:00:00       Robinia   
34222  CT19000859      Metro  6/15/2019 0:00:00       Robinia   
34223  CT19000860      Metro  6/15/2019 0:00:00       Robinia   
34224  CT19000861      Metro  6/15/2019 0:00:00       Robinia   
34225  CT19000862      Metro  6/15/2019 0:00:00       Robinia   
34226  CT19000863      Metro  6/15/2019 0:00:00       Robinia   
34227  CT19000864      Metro  6/15/2019 0:00:00      Taxodium   
34228  CT19000865      Metro  6/15/2019 0:00:00      Taxodium   
34229  CT19000866      Metro  6/15/2019 0:0

In [171]:
grouping2b = dbtree_sizes.groupby(['new_program'])
size_sum2b = grouping2b.describe()
size_sum2b

Year                                                 \
                       count         mean       std     min     25%     50%   
new_program                                                                   
Large Parcel          1527.0  2016.399476  2.976351  2012.0  2013.0  2019.0   
Other                16891.0  2015.736724  2.463640  2005.0  2014.0  2016.0   
Riversmart            4125.0  2011.958061  3.314421  2007.0  2010.0  2012.0   
Sustainable DC         647.0  2014.666151  0.890167  2014.0  2014.0  2014.0   
Tree Planting / CTP   9287.0  2008.813287  3.041182  2003.0  2006.0  2009.0   

                                     
                        75%     max  
new_program                          
Large Parcel         2019.0  2019.0  
Other                2018.0  2081.0  
Riversmart           2013.0  2019.0  
Sustainable DC       2015.0  2017.0  
Tree Planting / CTP  2011.0  2017.0

In [162]:
dbtree_sizes.to_csv(path_or_buf=r'E:/TreeSizes/Data/dbtree_sizes_812.csv', sep=',')

In [163]:
unknown = dbtree_sizes[dbtree_sizes['size'] == 'Unknown']
unknown_size_sciname = unknown.groupby(['SciName']).sum()
unknown_size_sciname.to_csv(path_or_buf=r'E:/TreeSizes/Data/unknown_sizes_812.csv', sep=',')